spec/models/song_spec.rb

```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+describe Song do
+  it { should validate_presence_of :title }
+  it { should validate_presence_of :artist }
+  it { should validate_presence_of :album }
+  it { should validate_presence_of :duration }
```

Comment:
Also if you added a uniqueness constraint to the DB, would want a matching validation here.